hydra-node/src/Hydra/Chain/Offline.hs

```diff
@@ -0,0 +1,138 @@
+module Hydra.Chain.Offline (
+  withOfflineChain,
+) where
+
+import Hydra.Prelude
+
+import Hydra.Chain.Offline.Handlers (mkFakeL1Chain)
+
+import Hydra.Logging (Tracer)
+
+import Hydra.Chain (
+  ChainComponent,
+  ChainEvent (Tick),
+  ChainStateHistory,
+  chainSlot,
+  chainTime,
+ )
+import Hydra.HeadId (HeadId)
+
+import Hydra.Chain.Direct.Handlers (
+  DirectChainLog (),
+  newLocalChainState,
+ )
+
+import Hydra.Ledger (ChainSlot (ChainSlot), IsTx (UTxOType))
+import Hydra.Ledger.Cardano.Configuration (readJsonFileThrow)
+
+import Hydra.Options (OfflineConfig (OfflineConfig, initialUTxOFile, ledgerGenesisFile))
+
+import Cardano.Ledger.Shelley.API qualified as Ledger
+
+import Ouroboros.Consensus.HardFork.History (interpretQuery, mkInterpreter, neverForksSummary, slotToWallclock, wallclockToSlot)
+import Ouroboros.Consensus.HardFork.History qualified as Consensus
+
+import Cardano.Ledger.Slot (SlotNo (SlotNo, unSlotNo))
+
+import Cardano.Slotting.Time (SystemStart (SystemStart), mkSlotLength, toRelativeTime)
+import Cardano.Slotting.Time qualified as Slotting
+
+import Cardano.Ledger.BaseTypes (epochInfoPure)
+
+import Cardano.Slotting.EpochInfo (EpochInfo (EpochInfo), epochInfoFirst, epochInfoSlotToUTCTime)
+
+import Ouroboros.Consensus.Util.Time (nominalDelay)
+
+import Hydra.Cardano.Api (
+  StandardCrypto,
+  Tx,
+ )
+import Hydra.Chain.Offline.Persistence (initializeStateIfOffline)
+import Hydra.ContestationPeriod (ContestationPeriod)
+import Hydra.Party (Party)
+
+withOfflineChain ::
+  Tracer IO DirectChainLog -> -- TODO(ELAINE): change type to indicate offline mode maybe?
+  OfflineConfig ->
+  Ledger.Globals ->
+  HeadId ->
+  Party ->
+  ContestationPeriod ->
+  -- | Last known chain state as loaded from persistence.
+  ChainStateHistory Tx ->
+  ChainComponent Tx IO a
+withOfflineChain tracer OfflineConfig{ledgerGenesisFile, initialUTxOFile} globals@Ledger.Globals{systemStart} ownHeadId party contestationPeriod chainStateHistory callback action = do
+  initialUTxO :: UTxOType Tx <- readJsonFileThrow (parseJSON @(UTxOType Tx)) initialUTxOFile
+  initializeStateIfOffline chainStateHistory initialUTxO ownHeadId party contestationPeriod callback
+
+  localChainState <- newLocalChainState chainStateHistory
+  let chainHandle = mkFakeL1Chain contestationPeriod localChainState tracer ownHeadId callback
```

Comment:
```suggestion
  let chainHandle =
        Chain
          { submitTx = const $ pure ()
          , draftCommitTx = \_ _ -> pure $ Left FailedToDraftTxNotInitializing
          , postTx = const $ pure ()
          }
```

If we add `OnCollectComTx` as commented in `initializeStateIfOffline`, we do not need to have a working `Chain` handle at all. We can just do nothing (or fail) if a user tries to `Close`.